### PR TITLE
(fixed) Refactor app/status.go to use structured logging #12723

### PR DESCRIPTION
**Pull request by r4zorgeek**
#### Summary

Refactoring app/status.go to use structured logging i.e. removing fmt.

#### Ticket Link

[MM-18280](https://mattermost.atlassian.net/projects/MM/issues/MM-18280?filter=allopenissues&orderby=priority%20DESC&keyword=app/status.go)

Fixes https://github.com/mattermost/mattermost-server/issues/12723

### CODE REVIEW

**Comment by r4zorgeek** [comment]
@mattermod @amyblais Hey I have refactored the code. Please review it.

**Comment by jasonblais** [comment]
@r4zorgeek Thank you for your contribution, appreciate it! Can you help review the build failure and update summary/ticket link for the PR? We'll then start the dev review.

If you have any questions on either, don't hesitate to let us know!

**Comment by r4zorgeek** [comment]
@jasonblais Hey, I looked at the build failure its stating gofmt faliure on line 305 of status.go. Please, can you help me to fix it? I couldn't understand the failure. I think it might be because of the styling issue.

**Comment by jasonblais** [comment]
Sure thing! Adding two dev reviewers who can help take a look at the build failure + review the PR. Thanks again for the contribution @r4zorgeek

**Comment by r4zorgeek** [comment]
@jasonblais fixed all the issues and also all test cases are passing. Please review it.

**Comment by r4zorgeek** [comment]
@reflog Changed "Failed to save status for" to "Failed to save status".

**Comment by reflog** [comment]
@r4zorgeek - thanks, but please replace all occurrences, not just the one

**Comment by r4zorgeek** [comment]
> @r4zorgeek - thanks, but please replace all occurrences, not just the one

Done

File: app/status.go [hunk 1]
```diff
@@ -215,11 +213,11 @@ func (a *App) SetStatusOnline(userId string, manual bool) {
 	if status.Status != oldStatus || status.Manual != oldManual || status.LastActivityAt-oldTime > model.STATUS_MIN_UPDATE_TIME {
 		if broadcast {
 			if err := a.Srv.Store.Status().SaveOrUpdate(status); err != nil {
-				mlog.Error(fmt.Sprintf("Failed to save status for user_id=%v, err=%v", userId, err), mlog.String("user_id", userId))
+				mlog.Error("Failed to save status", mlog.String("user_id=", userId), mlog.Err(err), mlog.String("user_id", userId))
```
Comment (hanzei) [inline app/status.go]: We don't need the `=`
```suggestion
				mlog.Error("Failed to save status", mlog.String("user_id", userId), mlog.Err(err), mlog.String("user_id", userId))
```

File: app/status.go [hunk 2]
```diff
@@ -215,11 +213,11 @@ func (a *App) SetStatusOnline(userId string, manual bool) {
...
 			}
 		} else {
 			if err := a.Srv.Store.Status().UpdateLastActivityAt(status.UserId, status.LastActivityAt); err != nil {
-				mlog.Error(fmt.Sprintf("Failed to save status for user_id=%v, err=%v", userId, err), mlog.String("user_id", userId))
+				mlog.Error("Failed to save status", mlog.String("user_id=", userId), mlog.Err(err), mlog.String("user_id", userId))
```
Comment (hanzei) [inline app/status.go]:
```suggestion
				mlog.Error("Failed to save status", mlog.String("user_id", userId), mlog.Err(err), mlog.String("user_id", userId))
```

**Comment by r4zorgeek** [comment]
@hanzei Welcome Sir.

**Comment by hanzei** [comment]
Would you please merge `master` into your branch?

**Comment by r4zorgeek** [comment]
@hanzei Hey sorry for late reply been busy lately. How should I merge with my branch? Sorry for asking this silly question, I'm new to opensource.

**Comment by hanzei** [comment]
Hi @r4zorgeek,

You can follow the documentation of GH:
https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/configuring-a-remote-for-a-fork and thenhttps://help.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork

Let me know if this helps.

**Comment by hanzei** [comment]
@r4zorgeek If you want, I can take care of the merge.

**Comment by r4zorgeek** [comment]
@hanzei Sorry Sir for the inconvenience caused. Been busy with college stuff. I'll finish it by Sunday.

**Comment by hanzei** [comment]
I've merged `master` for you @r4zorgeek. I hope you don't mind this. Thanks for the PR :+1:

**Pull request merged**